go/src/client/lc0_main.go

```diff
@@ -0,0 +1,487 @@
+			return "", ""
+		}
+		fmt.Fprintf(settingsFile, "%s", jsonSettings)
+		return settings.User, settings.Pass
+	}
+	defer file.Close()
+	decoder := json.NewDecoder(file)
+	err = decoder.Decode(&settings)
+	if err != nil {
+		log.Fatal("Error decoding JSON ", err)
+		return "", ""
+	}
+	return settings.User, settings.Pass
+}
+
+func getExtraParams() map[string]string {
+	return map[string]string{
+		"user":     *user,
+		"password": *password,
+		"version":  "10",
+	}
+}
+
+func uploadGame(httpClient *http.Client, path string, pgn string, nextGame client.NextGameResponse, version string, retryCount uint) error {
+	return nil
+	
+	extraParams := getExtraParams()
+	extraParams["training_id"] = strconv.Itoa(int(nextGame.TrainingId))
+	extraParams["network_id"] = strconv.Itoa(int(nextGame.NetworkId))
+	extraParams["pgn"] = pgn
+	extraParams["engineVersion"] = version
+	request, err := client.BuildUploadRequest(*hostname+"/upload_game", extraParams, "file", path)
+	if err != nil {
+		return err
+	}
+	resp, err := httpClient.Do(request)
+	if err != nil {
+		return err
+	}
+	body := &bytes.Buffer{}
+	_, err = body.ReadFrom(resp.Body)
+	if err != nil {
+		log.Print(err)
+		log.Print("Error uploading, retrying...")
+		time.Sleep(time.Second * (2 << retryCount))
+		err = uploadGame(httpClient, path, pgn, nextGame, version, retryCount+1)
+		return err
+	}
+	resp.Body.Close()
+	fmt.Println(resp.StatusCode)
+	fmt.Println(resp.Header)
+	fmt.Println(body)
+
+	train_dir := filepath.Dir(path)
+	if _, err := os.Stat(train_dir); err == nil {
+		files, err := ioutil.ReadDir(train_dir)
+		if err != nil {
+			log.Fatal(err)
+		}
+		fmt.Printf("Cleanup training files:\n")
+		for _, f := range files {
+			fmt.Printf("%s/%s\n", train_dir, f.Name())
+		}
+		err = os.RemoveAll(train_dir)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	return nil
+}
+
+type gameInfo struct {
+	pgn   string
+	fname string
+}
+
+type CmdWrapper struct {
+	Cmd      *exec.Cmd
+	Pgn      string
+	Input    io.WriteCloser
+	BestMove chan string
+	gi       chan gameInfo
+	Version  string
+}
+
+func (c *CmdWrapper) openInput() {
+	var err error
+	c.Input, err = c.Cmd.StdinPipe()
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func convertMovesToPGN(moves []string) string {
+	game := chess.NewGame(chess.UseNotation(chess.LongAlgebraicNotation{}))
+	for _, m := range moves {
+		game.MoveStr(m)
+	}
+	return game.String()
+}
+
+func CreateCmdWrapper() *CmdWrapper {
+	c := &CmdWrapper{
+		gi: make(chan gameInfo),
+		BestMove: make(chan string),
+	}
+	return c
+}
+
+func (c *CmdWrapper) launch(networkPath string, args []string, input bool) {
+	weights := fmt.Sprintf("--weights=%s", networkPath)
+	dir, _ := os.Getwd()
+	c.Cmd = exec.Command(path.Join(dir, "lc0"), "selfplay", weights, "--training=true", "--visits=800")
+	// TODO: server args are hosed right now.
+//	c.Cmd.Args = append(c.Cmd.Args, args...)
+	if *gpu != -1 {
+		c.Cmd.Args = append(c.Cmd.Args, fmt.Sprintf("--gpu=%v", *gpu))
```

Comment:
There is no -gpu flag in `lc0`. Maybe it's possible just to forward all unknown flags to `lc0`? Or maybe some other way... There is --parallelism, backend configuration, and other stuff..

Reply:
done